### PR TITLE
ports to master (part 1)

### PR DESCRIPTION
**Pull request by Dikay900**
referencing PRs #81 #3923 #3824 #3721

edit: reordered PRs to commit list

### CODE REVIEW

**Comment by Dikay900** [comment]
added port for #3923 #3824

**Comment by Dikay900** [comment]
if referenced people are active please review your own commits, if someone find a bug please give me a feedback.

**Comment by StevenPuttemans** [comment]
Hmm great! Just too bad this doesn't happen automatically anymore ...
Keep up the good work!

**Comment by Dikay900** [comment]
y most of the changes have to be slightly changed to get into the other branch so this make simple automatic merging unfortunately impossible

File: samples/cpp/openni_capture.cpp [hunk 1]
```diff
@@ -13,14 +13,14 @@ static void help()
...
+            "   CAP_OPENNI_POINT_CLOUD_MAP      - XYZ in meters (CV_32FC3)\n"
+            "   CAP_OPENNI_DISPARITY_MAP        - disparity in pixels (CV_8UC1)\n"
+            "   CAP_OPENNI_DISPARITY_MAP_32F    - disparity in pixels (CV_32FC1)\n"
+            "   CAP_OPENNI_VALID_DEPTH_MASK     - mask of valid pixels (not ocluded, not shaded etc.) (CV_8UC1)\n"
             "2.) Data given from RGB image generator\n"
-            "   CV_CAP_OPENNI_BGR_IMAGE            - color image (CV_8UC3)\n"
-            "   CV_CAP_OPENNI_GRAY_IMAGE           - gray image (CV_8UC1)\n"
+            "   CAP_OPENNI_BGR_IMAGE            - color image (CV_8UC3)\n"
+            "   CAP_OPENNI_GRAY_IMAGE           - gray image (CV_8UC1)\n"
          << endl;
```
Comment (StevenPuttemans) [inline samples/cpp/openni_capture.cpp]: Actually I am not sure if 2.4 made a complete switch to not using the CV_ prefix. I would be cautious with simple removing them!

Comment (Dikay900) [inline samples/cpp/openni_capture.cpp]: and this is master ^^ sorry for confusing with my branch name will correct this in the future ^^
```
into Itseez:master
```

Comment (StevenPuttemans) [inline samples/cpp/openni_capture.cpp]: :+1:

**Comment by vpisarev** [comment]
:+1:

**Pull request merged**